jackson-merge-patch-validations/src/test/java/com/github/jeffnelson/jackson/patch/validator/PatchNumberValidationsControllerTest.java

```diff
@@ -0,0 +1,243 @@
+        String content = getFileContent("json/invalidNumbers.json");
+
+        mvc.perform(patch("/foo")
+                .content(content)
+                .contentType(PatchMediaType.APPLICATION_MERGE_PATCH_JSON))
+                .andExpect(status().is4xxClientError())
+                .andExpect(jsonPath("[0].field", is("myDbl")))
+                .andExpect(jsonPath("[0].error", is("must be less than or equal to 50")))
+                .andExpect(jsonPath("[1].field", is("myLong")))
+                .andExpect(jsonPath("[1].error", is("Required")))
+                .andExpect(jsonPath("[2].field", is("myInt")))
+                .andExpect(jsonPath("[2].error", is("must be greater than or equal to 12")))
+                .andReturn();
+
+        assertNull(controller.foo);
+    }
+
+    void assertTestDTO(
+            TestDTO actual,
+            PatchField<Integer> expectedMyInt,
+            PatchField<Double> expectedMyDbl,
+            PatchField<Long> expectedMyLong) {
+        assertTestDTO("", actual, expectedMyInt, expectedMyDbl, expectedMyLong);
+    }
+
+    void assertTestDTO(String fieldNamePrefix,
+            TestDTO actual,
+            PatchField<Integer> expectedMyInt,
+            PatchField<Double> expectedMyDbl,
+            PatchField<Long> expectedMyLong) {
+        assertPatchField(combine(fieldNamePrefix, "myInt"), expectedMyInt, actual.getMyInt());
+        assertPatchField(combine(fieldNamePrefix, "myDbl"), expectedMyDbl, actual.getMyDbl());
+        assertPatchField(combine(fieldNamePrefix, "myLong"), expectedMyLong, actual.getMyLong());
+    }
+
+    <T> void assertPatchField(String fieldName, PatchField<T> expected, PatchField<T> actual) {
+        assertEquals(combine(fieldName, " shouldPatch incorrect"), expected.shouldPatch(), actual.shouldPatch());
+        assertEquals(combine(fieldName, " value incorrect"), expected.getValue(), actual.getValue());
+    }
+
+    String combine(String prefix, String name) {
+        return String.format("%s%s", prefix, name);
+    }
+
+    @RestController
+    public static class TestRestController {
+
+        FooDTO foo;
+
+        @RequestMapping(value = "/foo", method = RequestMethod.PATCH, consumes = PatchMediaType.APPLICATION_MERGE_PATCH_JSON)
+        public Void patchFoo(@Valid @RequestBody FooDTO foo) {
+            this.foo = foo;
+            return null;
+        }
+
+    }
+
+    @ControllerAdvice
+    @RequiredArgsConstructor
+    public static class WebExceptionHandler extends ResponseEntityExceptionHandler {
+
+        final MessageSource messageSource;
+
+        Exception caught;
+
+        // @ExceptionHandler({Exception.class})
+        // public ResponseEntity<Object> handleUnknownException(Exception ex, WebRequest request) throws Exception {
+        // caught = ex;
+        // return new ResponseEntity<>(String.format("unhandled exception: %s", ex.getMessage()),
+        // HttpStatus.BAD_REQUEST);
+        // }
```

Comment:
Is this still needed / planned to be uncommented? Or can it be deleted?

Reply:
pushed